presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -973,7 +975,26 @@ protected Object visitArrayConstructor(ArrayConstructor node, Object context)
         @Override
         protected Object visitRow(Row node, Object context)
         {
-            throw new PrestoException(NOT_SUPPORTED, "Row expressions not yet supported");
+            RowType rowType = (RowType) expressionTypes.get(node);
+            List<Type> parameterTypes = rowType.getTypeParameters();
+            List<Expression> arguments = node.getItems();
+
+            int n = arguments.size();
```

Comment:
We try to avoid single letter variable names, so either inline this or name it something more descriptive like "cardinality"